code/game/gamemodes/changeling/powers/linglink.dm

```diff
@@ -0,0 +1,67 @@
+/obj/effect/proc_holder/changeling/linglink
+	name = "Hivemind Link"
+	desc = "Link your victim's mind into the hivemind for personal interrogation"
+	chemical_cost = 0
+	dna_cost = 0
+	req_human = 1
+	max_genetic_damage = 100
+
+/obj/effect/proc_holder/changeling/linglink/can_sting(mob/living/carbon/user)
+	if(!..())
+		return
+	var/datum/changeling/changeling = user.mind.changeling
+	var/obj/item/weapon/grab/G = user.get_active_hand()
+	var/mob/living/carbon/target = G.affecting
+	if(changeling.islinking)
+		user << "<span class='warning'>We have already formed a link with the victim!</span>"
+		return
+	if(!target.mind)
+		user << "<span class='warning'>The victim has no mind to link to!</span>"
+		return
+	if(target.stat == DEAD)
+		user << "<span class='warning'>The victim is dead, you cannot link to a dead mind!</span>"
+		return
+	if(target.mind.changeling)
+		user << "<span class='warning'>The victim is already a part of the hivemind!</span>"
+		return	
+	if(!istype(G))
+		user << "<span class='warning'>We must be tightly grabbing a creature in our active hand to link with them!</span>"
+		return
+	if(G.state <= GRAB_NECK)
+		user << "<span class='warning'>We must have a tighter grip to link with this creature!</span>"
+		return
+	return changeling.can_absorb_dna(user,target)
+
+/obj/effect/proc_holder/changeling/linglink/sting_action(mob/user)
+	var/datum/changeling/changeling = user.mind.changeling
+	var/obj/item/weapon/grab/G = user.get_active_hand()
+	var/mob/living/carbon/human/target = G.affecting
+	changeling.islinking = 1
+	for(var/stage = 1, stage<=3, stage++)
+		switch(stage)
+			if(1)
+				user << "<span class='notice'>This creature is compatible. We must hold still...</span>"
+			if(2)
+				user << "<span class='notice'>We stealthily stab [target] with a minor proboscis...</span>"
+				target << "<span class='userdanger'>You experience a stabbing sensation and your ears begin to ring...</span>"
+			if(3)
+				user << "<span class='notice'>You mold the [target]'s mind like clay, they can now speak in the hivemind!</span>"
+				target << "<span class='userdanger'>A migraine throbs behind your eyes, you hear yourself screaming - but your mouth has not opened!</span>"
+				for(var/mob/M in mob_list)
+					if(M.lingcheck() == 2)
```

Comment:
this is bad but not your fault really.
If you're feeling up to it I'd like the values from ```lingcheck()``` to be ```#define```s, but it's not required.